### ISSUE TITLE
Search filters for Multiple ManyToOne properties (as in "IN (X, Y)") return incorrect results

### ISSUE THREAD

**Issue by yivi** [api-platform/api-platform#1537]
I've tried this with Api-Core 2.5.5, running on PHP 7.4.5 and 7.4.6, over Symfony 5.0.8

**Description**  
This is a weird.

```
GET /api/reviews?book[]=/api/books/2&book[]=/api/books/3
```

should return all the reviews linked to `Book#2`, and `Book#3`. Instead, it returns the reviews for `Book#1`.

E.g.:
Attempting:
<img width="541" alt="imagen" src="https://user-images.githubusercontent.com/1815039/83033064-e93d2c80-a036-11ea-8e40-4381868e81be.png">

```
http://localhost:7099/api/reviews?book%5B%5D=%2Fapi%2Fbooks%2F3&book%5B%5D=%2Fapi%2Fbooks%2F2&page=1
```

Results in:
<img width="817" alt="imagen" src="https://user-images.githubusercontent.com/1815039/83033118-fd812980-a036-11ea-8810-c8ecc63b5639.png">

**How to reproduce**  
I've created a repo here https://github.com/yivi/api_platform_filter_problem_mre demonstrating the issue.

The repo was created by simply creating a symfony-skeleton project and, plus `require api`, and copying the `Book` and `Review` entities from the documentation (with the addition of the `ApiFilter` annotation).

Here is a sample sqlite db, with 3 books and 4 reviews. [app.db.zip](https://github.com/api-platform/api-platform/files/4689373/app.db.zip)

**Possible Solution**  
Wish I knew. As far as I can see on the debugger, the correct queries are being generated:
<img width="1071" alt="imagen" src="https://user-images.githubusercontent.com/1815039/83033611-9e6fe480-a037-11ea-8365-88101a50be55.png">

And yet, the results are very clearly wrong. 

I'd love to help with this issue, but I'm not sure where exactly this is going wrong, so I would need at least a pointer to start debugging in the right place.

**Comment by yivi** [comment]
Using the same search syntax but for a single id works as expected:


```
GET http://localhost:7099/api/reviews?book[]=/api/books/3&page=1
```
<img width="603" alt="imagen" src="https://user-images.githubusercontent.com/1815039/83103838-cacb4580-a0b7-11ea-96b8-8b8ac75850c7.png">

or

```
GET http://localhost:7099/api/reviews?book[]=/api/books/2&page=1
```

<img width="730" alt="imagen" src="https://user-images.githubusercontent.com/1815039/83103966-1382fe80-a0b8-11ea-8fa1-3c9da5e6ec6b.png">


But trying the one earlier (`GET http://localhost:7099/api/reviews?book[]=/api/books/2&book[]=/api/books/3`, or any variation) will include in the results only the reviews that reference `Book#1`

**Comment by yivi** [comment]
...  aaaand it was fixed in 2.5.6 👍
